packages/gui/src/components/offers2/OfferBuilderViewer.tsx

```diff
@@ -72,6 +79,31 @@ function OfferBuilderViewer(props: OfferBuilderViewerProps, ref: any) {
 
   const showInvalid = !isValidating && isValid === false;
 
+  const hasExpiration = offerSummary.validTimes?.maxTime !== undefined && offerSummary.validTimes?.maxTime !== 0;
+  let currentTime = null;
+  let expirationTime = null;
+  let isExpired = false;
+
+  const { data: height, isLoading: isGetHeightInfoLoading } = useGetHeightInfoQuery(undefined, {
+    pollingInterval: 3000,
+  });
+  const { data: lastBlockTimeStampData, isLoading: isGetTimestampForHeightLoading } = useGetTimestampForHeightQuery({
+    height: height || 0,
+  });
+
+  if (hasExpiration && !isExpired) {
+    const lastBlockTimeStamp = lastBlockTimeStampData?.timestamp || 0;
+    const currentTimeMoment = moment.unix(lastBlockTimeStamp - 20);
```

Comment:
Since I've seen the same code 3 times now, I think we should have a common function. Duplicating code twice is already one time too many. Please take a look at the lines that are being duplicated and figure out a way to refactor the code so that we only have a single implementation.